src/darwin/Framework/Matter.xcodeproj/project.pbxproj

```diff
@@ -262,6 +262,8 @@
 		7596A8512878709F004DAE0E /* MTRAsyncCallbackQueueTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 7596A8502878709F004DAE0E /* MTRAsyncCallbackQueueTests.m */; };
 		7596A85528788557004DAE0E /* MTRClusters.mm in Sources */ = {isa = PBXBuildFile; fileRef = 7596A85228788557004DAE0E /* MTRClusters.mm */; };
 		7596A85728788557004DAE0E /* MTRClusters.h in Headers */ = {isa = PBXBuildFile; fileRef = 7596A85428788557004DAE0E /* MTRClusters.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		75A202E52BA8DBAC00A771DD /* reporting.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 75A202E42BA8DBAC00A771DD /* reporting.cpp */; };
```

Comment:
So this can't be part of libCHIP because it uses emberAfDataVersionStorage which depends on generated stuff?

Reply:
I got a hit on `ember-compatibility-functions.cpp` inside this file, so I added it.

The reference seems to be at line 154 above here. Since I moved files from ember-compatibility-functions.cpp, I thought reporting.cpp has to reside here as well.

What is the correct setup here?

Reply:
The correct setup is that if reporting.cpp is built as part of libCHIP (which ember-compatibility-functions.cpp is not), then it should not be here, and it it's not built as part of libCHIP then it needs to be here, looks like.

The real question is whether reporting.cpp could be built as part of libCHIP.

Reply:
But the other comments explain why it can't be, so we are all good here.